src/libraries/System.Private.CoreLib/src/System/IO/File.cs

```diff
@@ -251,27 +251,33 @@ public static void WriteAllText(string path, string? contents, Encoding encoding
 
         public static byte[] ReadAllBytes(string path)
         {
-            // bufferSize == 1 used to avoid unnecessary buffer in FileStream
-            using (FileStream fs = new FileStream(path, FileMode.Open, FileAccess.Read, FileShare.Read, bufferSize: 1, FileOptions.SequentialScan))
+            // SequentialScan is a perf hint that requires extra sys-call on non-Windows OSes.
+            FileOptions options = OperatingSystem.IsWindows() ? FileOptions.SequentialScan : FileOptions.None;
```

Comment:
nit: maybe turn this into a static property, like `PlatformSequentialScanOption`.